java/arcs/android/sdk/host/AndroidResurrector.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.android.sdk.host
+
+import android.app.Service
+import arcs.core.host.Resurrector
+import arcs.core.host.ResurrectorBase
+import arcs.core.storage.StorageKey
+import arcs.sdk.android.storage.ResurrectionHelper
+
+/**
+ * An implementation of [Resurrector] that is backed by [AndroidResurrector]. This implementation
+ * currently has limitations, in that it does not persist the arcId to storageKey mappings, but
+ * uses memory resident maps. This is a workaround until [ResurrectorService] is upgraded to
+ * support this feature.
+ */
+open class AndroidResurrector(val context: Service) : ResurrectorBase() {
```

Comment:
Why is the parameter `Service` type? `ResurrectionHelper` only needs a generic `Context` right?

If this can be changed to `AndroidResurrector(val context: Context)`, then `AndroidHost` can instantiate an instance using the `context` passed in, instead of also needing an instance as a parameter.

Reply:
No, it needs to be the Service that embeds the ArcHost. The Context passed to the ResurrectionHelper is used as a class name to construct a ComponentName used to send the return intents to wake you up.  Technically, it could be an Activity, but currently the codebase doesn't let Activities embed ArcHosts. This could be rectified in ArcHostHelper and AndroidManifestHostRegistry

Reply:
I see now. Thanks for pointing me in the right direction on this.